### PR TITLE
Generic param overloads

### PR DESCRIPTION
**Pull request by mikaelm12**
Added overloads so that users can use up to 7 parameters.
I'll rebase and add tests when https://github.com/aspnet/SignalR/pull/52 is merged into the net-client branch.
@anurse @moozzyk @davidfowl @BrennanConroy

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs [hunk 1]
```diff
@@ -47,6 +47,41 @@ public void On(string methodName, Type[] parameterTypes, Action<object[]> handle
             _handlers.AddOrUpdate(methodName, invocationHandler, (_, __) => invocationHandler);
         }
 
+        public void On<T1>(string methodName, T1 parameter1, Action<object[]> handler)
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs]: I don't think these should be methods on this class. Rather I'd prefer these to be extension methods.

File: src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs [hunk 2]
```diff
@@ -47,6 +47,41 @@ public void On(string methodName, Type[] parameterTypes, Action<object[]> handle
...
+        {
+            On(methodName, new[] { typeof(T1) }, handler);
+        }
+
+        public void On<T1,T2>(string methodName, T1 parameter1, T2 parameter2, Action<object[]> handler)
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs]: nit: space between T1 and T2

File: src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs (hunk 1 above)
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs]: Previously they were extensions on IHubProxy - https://github.com/SignalR/SignalR/blob/dev/src/Microsoft.AspNet.SignalR.Client/HubProxyExtensions.cs

**Comment by davidfowl** [comment]
@mikaelm12 we shouldn't be doing PRs on top of non master branches. I'd like to get the basic client in with some tests then build on that.

**Comment by mikaelm12** [comment]
Cool. Noted.

**Pull request closed without merging**